website/guides/sgb_border.md

```diff
@@ -0,0 +1,178 @@
+
+- It would be a good idea to save somewhere in RAM whether the game is running on an SGB capable device or not, such that if you wish to change the border mid-gameplay, you won't have to perform SGB detection again
+
+- **If you wish to only use 1 controller for the game, you will have to send another `MLT_REQ` to disable multiplayer** (`$89, $00`)
+
+- You can change the waitloop to not use interrupts, or to use di+halt instead
+
+## Border limitations
+
+An SGB border has:
+
+- 255 tiles + 1 transparent tile (preferably tile #0)
+- 3 palettes of 15 (+ 1 transparent) colors, (up to 45 solid colors total)
+- a 256x224px tilemap (there's a bit more to this, see [notes](#notes))
+
+## Converting borders
+
+With a recent version of [superfamiconv](https://github.com/Optiroc/SuperFamiconv):
+
+```sh
+superfamiconv -v -i input.png -p output.pal -t output.4bpp -P 4 -m output.pct -M snes --color-zero 0000ff -B 4
+```
+
+- `--color-zero` should be the color that your image for transparency, in my case it was blue.
+	* If your image has an alpha channel, it can be set can also be set to `00000000` to use the actual transparent color; however, this may cause some issues.
+- `-v` is optional, for showing details of the conversion process
+- You can add a row of the transparent color at the top of the image to force superfamiconv to make it tile #0, then `incbin "output.pct", 64` to leave out that row.
+- `-P 4` sets the base palette to the 4th one, and **SGB borders use SNES palettes 4, 5, and 6.** as of writing this, this option only works if you built superfamiconv from source.
+
+## Uploading borders
+
+As stated before, the SGB border consists of tile data, picture data, and palette data. These are split across 2-3 packets:
+
+- `CHR_TRN` (`$99`) is used to send 4KiB of tile data.
+
+	* since the border can use up to 8KiB of tiles, bit 0 of the second byte specifies which "half" you're sending
+		- `$99, $00` if the screen is loaded with the first 4KiB of tile data
+		- `$99, $01` if the screen is loaded with the second 4KiB of tile data
+
+- `PCT_TRN` (`$a1`) is used to send the picture and palette data. it also swaps the border, generally a good idea to send it after the tile data[^1]
+
+	* assuming tiles `0`-`255` use VRAM from `$8000` to `$8fff`:
+		- the picture data must be at `$8000`-`$873f` (last 64 bytes are _usually_ offscreen, see [notes](#notes))
+		- palette data must be at `$8800`-`$885f`
+		- everything else is ignored
+			* how you skip putting data at `$8740`-`$87ff` is up to you, I prefer doing separate copies, valen prefers copying tilemap and palette data in one go, with the area between them pa
+
+See also the related Pan Docs entry: [SGB Command Border](https://gbdev.io/pandocs/SGB_Command_Border.html).
+
+[^1]: You can send a `CHR_TRN` up to \~60 frames after the `PCT_TRN` for it to apply to the current border, but not all emulators will emulate this. its fine to just pretend `CHR_TRN`s must go before `PCT_TRN`
```

Comment:
```suggestion
[^1]: You can send a `CHR_TRN` [up to \~60 frames](https://github.com/pinobatch/little-things-gb/blob/master/sgbears/docs/long_story.txt) after the `PCT_TRN` for it to apply to the current border, but not all emulators will emulate this. It's fine to just pretend `CHR_TRN`s must go before `PCT_TRN`.
```

Reply:
I might change the name of `long_story.txt` in a future revision of this repository. Sometimes I leave things as `.txt` while writing them until I finish enough of the first draft to be able to tell whether or not I'll be using features that make the Markdown source code less readable in a text editor. Should this tutorial link to `long_story.txt` in a particular revision, or should it wait until I've made a decision about renaming it?

Reply:
The link right now it's broken for me. We should link to it in a particular git commit so the link won't break and you can PR the new link if/when it changes.

Reply:
Link to a specific commit <https://github.com/pinobatch/little-things-gb/blob/b11b554d73c48a0f54fee0df31e59eb83806fcb4/sgbears/docs/long_story.md>

Reply:
Thanks Pino! Updated my original suggestion accordingly.